subworkflows/local/hic_minimap2.nf

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env nextflow
+
+// This subworkflow takes an input fasta sequence and csv style list of hic cram file to return
+// alignment files including .mcool, pretext and .hic.
+// Input - Assembled genomic fasta file, cram file directory
+// Output - .mcool, .pretext, .hic
+
+//
+// MODULE IMPORT BLOCK
+//
+include { CRAM_FILTER_MINIMAP2_FILTER5END_FIXMATE_SORT    } from '../../modules/local/cram_filter_minimap2_filter5end_fixmate_sort'
+include { SAMTOOLS_MERGE                                  } from '../../modules/nf-core/samtools/merge/main'
+include { MINIMAP2_INDEX                                  } from '../../modules/nf-core/minimap2/index/main'
+
+
+workflow HIC_MINIMAP2 {
+
+    take:
+    reference_tuple     // Channel: tuple [ val(meta), path( file )      ]
+    csv_ch
+    reference_index
+
+    main:
+    ch_versions         = Channel.empty()
+    mappedbam_ch        = Channel.empty()
+
+    //
+    // MODULE: generate minimap2 mmi file
+    //
+    MINIMAP2_INDEX (
+        reference_tuple
+        )
+    ch_versions         = ch_versions.mix( MINIMAP2_INDEX.out.versions )
+
+    //
+    // LOGIC: generate input channel for mapping
+    //
+    csv_ch
+        .splitCsv()
+        .combine ( reference_tuple )
+        .combine ( MINIMAP2_INDEX.out.index )
+        .map{ cram_id, cram_info, ref_id, ref_dir, mmi_id, mmi_path->
+            tuple([
+                    id: cram_id.id
+                    ],
+                file(cram_info[0]),
```

Comment:
Where can I see an explanation of this array ? Maybe add a comment to indicate where the array is defined, and if the code is not straightforward, add some explanations there

Reply:
Updated to contain

```
     //
    // LOGIC:   generate input channel for mapping
    //          csv_ch is generated by generate_cram_csv found in line 42 of the GENERATE_MAPS subworkflow
    //          cram_info = [$crampath ${crampath}.crai, ${from} (container number), ${to} (container number), ${base}, ${chunkn}, ${rgline}]
    //
```